components/server/ee/src/workspace/gitpod-server-impl.ts

```diff
@@ -2123,33 +2123,51 @@ export class GitpodServerEEImpl extends GitpodServerImpl {
         }
     }
 
-    protected defaultSpendingLimit = 100;
-    async subscribeTeamToStripe(ctx: TraceContext, teamId: string, setupIntentId: string): Promise<void> {
-        this.checkAndBlockUser("subscribeUserToStripe");
-        const team = await this.guardTeamOperation(teamId, "update");
-        await this.ensureStripeApiIsAllowed({ team });
+    async subscribeToStripe(ctx: TraceContext, attributionId: string, setupIntentId: string): Promise<void> {
+        const attrId = AttributionId.parse(attributionId);
+        if (attrId === undefined) {
+            log.error(`Invalid attribution id: ${attributionId}`);
+            throw new ResponseError(ErrorCodes.BAD_REQUEST, `Invalid attibution id: ${attributionId}`);
+        }
+
+        const user = this.checkAndBlockUser("subscribeToStripe");
+
+        let customer: Stripe.Customer | undefined;
         try {
-            let customer = await this.stripeService.findCustomerByTeamId(team!.id);
+            if (attrId.kind === "team") {
+                const team = await this.guardTeamOperation(attrId.teamId, "update");
+                await this.ensureStripeApiIsAllowed({ team });
+                customer = await this.stripeService.findCustomerByTeamId(team!.id);
+            } else {
+                await this.ensureStripeApiIsAllowed({ user });
+                customer = await this.stripeService.findCustomerByUserId(user.id);
+            }
             if (!customer) {
-                throw new Error(`No Stripe customer profile for team '${team.id}'`);
+                throw new Error(`No Stripe customer profile for '${attributionId}'`);
             }
+
             await this.stripeService.setDefaultPaymentMethodForCustomer(customer, setupIntentId);
             await this.stripeService.createSubscriptionForCustomer(customer);
-
-            const attributionId: AttributionId = { kind: "team", teamId };
-
-            // Creating a cost center for this team
             await this.usageServiceClientProvider.getDefault().setCostCenter({
-                id: attributionId,
+                id: attrId,
                 spendingLimit: this.defaultSpendingLimit,
                 billingStrategy: "stripe",
             });
         } catch (error) {
-            log.error(`Failed to subscribe team '${teamId}' to Stripe`, error);
-            throw new ResponseError(ErrorCodes.INTERNAL_SERVER_ERROR, `Failed to subscribe team '${teamId}' to Stripe`);
+            log.error(`Failed to subscribe '${attributionId}' to Stripe`, error);
+            throw new ResponseError(
+                ErrorCodes.INTERNAL_SERVER_ERROR,
+                `Failed to subscribe '${attributionId}' to Stripe`,
+            );
         }
     }
 
+    protected defaultSpendingLimit = 100;
```

Comment:
Is this still true? https://github.com/gitpod-io/gitpod/pull/12882/files#diff-0caf79924a209df9577c140a9aefa82d74dab19ed1d20bd1f8597259c3d83316R48 seems to be using 0 as a value as well.

Reply:
have left a comment on that PR.

Reply:
This is an ok default spending limit for stripe-backed costCenters (ok, because it would be better to propose something reasonable based on past usage and/or number of users in a team).
For general costCenters we should assume 0, or better it should never be undefined on cost centers.